code/espurna/relay.ino

```diff
@@ -719,39 +719,23 @@ String _relayFriendlyName(unsigned char i) {
     return res;
 }
 
-void _relayWebSocketSendRelays(JsonObject& root) {
-    JsonObject& relays = root.createNestedArray("relays");
-
-
-    JsonArray& gpio = relays.createNestedArray("gpio");
-    JsonArray& type = relays.createNestedArray("type");
-    JsonArray& reset = relays.createNestedArray("reset");
-    JsonArray& boot = relays.createNestedArray("boot");
-    JsonArray& pulse = relays.createNestedArray("pulse");
-    JsonArray& pulse_time = relays.createNestedArray("pulse_time");
 
-    #if MQTT_SUPPORT
-        JsonArray& group = relays.createNestedArray("group");
-        JsonArray& group_sync = relays.createNestedArray("group_sync");
-        JsonArray& on_disconnect = relays.createNestedArray("on_disc");
-    #endif
+void _relayWebSocketSendRelays(JsonObject& root) {
+    JsonArray& relays = root.createNestedArray("relays");
```

Comment:
Note for this change and further structs as objects, that this was intentional: https://github.com/xoseperez/espurna/issues/1367, #1614 #1387 #1843 
Each structure adds up in size, both memory for storage and for transfer buffers

Reply:
Mmm, I was still thinking about this design part, whether the UI should compute the data or the websocket send a cleaner output, I believe a cleaner output is the way to go

About the memory, isn't the buffer cleared after the json message is sent?
It should actually not take a significant amount more (It's btw just to send it to the websocket and not for EEPROM storage), storage won't be modified (at least not in this PR)

Reply:
My main issue is the data received is not easily readable/iterable, the root should be clustered by modules instead of having each module intersect

Reply:
JSON buffer would only keep the data until it is stringified, then the string itself needs to be kept until the network accepts all of the data. RFBridge is a good case where there are a lot of relays at the same time, but you can add them on any board by using DUMMY_RELAY_COUNT=N

No problem using a key per module even when they are not all sent in the same message, but nesting things and having a lot of objects is not trivial for memory. 
But is there any specific reason for avoiding property=[n1,n2,n3] as a way to compose the data? Mentioning rfbridge again, there is also a way to partially send the data (via start_index and size values)

Reply:
And to get estimates on size without running any code:
https://arduinojson.org/v5/assistant/ (we are currently using this one)
https://arduinojson.org/v6/assistant/ (slightly larger than v5, so there's a hold up here)

Reply:
I'll keep the start_index in mind for partial data transfer, I did some calculations with the link provided, the code adds 50 bytes of memory per relay (minus one relay), is it gonna be a problem?

The reason for preferencing [{property:n1,id:1},{property:n2,id:2},{property:n3,id3}] over {property:[n1,n2,n3],id:[1,2,3]} is OOP and data compartmentalization, my teachers used to talk a big deal about this when I learned C, about how it's dangerous to have data representing an object sparsed into arrays (If you delete or add an item in an array but not the other you can break the entire representation, so you need to make sure you add/delete in every arrays, whereas with the other representation if you delete or add an item, well there is no problem)

So sure it adds a bit of memory because you'll have to duplicate the keys multiple times but in the long run if we start adding features, it's less prone to bugs and mistakes and also easier to parse, edit and display properties on the other side

Reply:
For example: https://gist.github.com/mcspr/63f5213f425fa9af95cf60f893c755c3
More around 1...1.5KB, the resulting string grows 3 times in size. Group string is probably needless, just an example. You can also prepare sensor example with multiple magnitudes, something like HLW8012, which are probably a second most common case with a lot of data.
Issues with memory blowing up are rather annoying things to have :/

The case of missing values or size mismatch could be enforced by the sender, we control both sides. However there may be an issue of having too much things in relay object itself or any other object where we need to have a lot of such arrays for properties.

So it feels rather backwards to just undo this optimization (note the PR! this was the way Xose implemented it initially, barring the namespace keys you are introducing and updated queueing mechanism).
Perhaps the thing we want is to send objects as arrays i.e. ctor arguments and prepend some kind of schema? See last json in the gist